compiler/rustc_middle/src/query/on_disk_cache.rs

```diff
@@ -864,7 +861,7 @@ impl_ref_decoder! {<'tcx>
 /// An encoder that can write to the incremental compilation cache.
 pub struct CacheEncoder<'a, 'tcx> {
     tcx: TyCtxt<'tcx>,
-    encoder: FileEncoder,
+    pub encoder: FileEncoder,
```

Comment:
where is this used? Could we instead expose a public method for doing whatever logic needs this encoder, or would that require a lot of methods?

Reply:
There was a trait method I could use instead.